### PR TITLE
Fixed `Digest::CRC5`

### PR DESCRIPTION
**Pull request by dearblue**
Previous implementations had the following problems:

  - The internal value was always calculated to be shifted three bits to the left.
  - The external output should have been shifted three bits to the right, but was not.
  - The C implementation was not bit-shifted.

As a result, the correct calculation was not being done.

With this patch, the Ruby implementation is now also not left bit-shifted, and the calculation is no longer a problem.

However, there are the following compatibility concerns:

  - When users have made their own modifications
  - When users refer to `Digest::CRC5::TABLE`.

### CODE REVIEW

**Comment by dearblue** [comment]
As a reminder, here is a brief summary of what is planned after this PR is merged:

  - All `#updates` will be merged into `Digest::CRC#update`, so the `Digest::CRC5#update` that this PR modifies will also be removed.
  - The `Digest::CRC#pack` will also be merged, so the separate `#pack` method will be removed.
  - The C implementation also provides a general-purpose `Digest::CRC#update` method. Thus, specialized so files will only have initialization functions and table data.

File: lib/digest/crc5.rb [hunk 1]
```diff
@@ -12,37 +12,26 @@ class CRC5 < CRC
 
     XOR_MASK = 0x1f
 
-    CRC_MASK = (0x1f << 3)
```
Comment (postmodern) [inline lib/digest/crc5.rb:15]: According to [pycrc-0.10.0](https://pycrc.org/)'s (reference implementation) generated C code, the CRC "mask" should be `0x1f`.

```shell
/src/pycrc.py --generate c -o crc5.c --model crc-5 --algorithm=table-driven
```

```c
crc_t crc_update(crc_t crc, const void *data, size_t data_len)
{
    const unsigned char *d = (const unsigned char *)data;
    unsigned int tbl_idx;

    while (data_len--) {
        tbl_idx = crc ^ *d;
        crc = (crc_table[tbl_idx] ^ (crc >> 8)) & 0x1f;
        d++;
    }
    return crc & 0x1f;
}
```

File: lib/digest/crc5.rb [hunk 2]
```diff
@@ -64,7 +53,7 @@ def self.pack(crc)
     #
     def update(data)
       data.each_byte do |b|
-        @crc = ((@table[((@crc >> 3) ^ b) & 0xff] ^ (@crc >> 8)) & @crc_mask)
+        @crc = @table[(@crc ^ b) & 0xff] ^ (@crc >> 8)
```
Comment (postmodern) [inline lib/digest/crc5.rb:56]: The C code which [pycrc-0.10.0](https://pycrc.org/) (reference implementation) generated still uses a CRC mask of `0x1f`.

```shell
/src/pycrc.py --generate c -o crc5.c --model crc-5 --algorithm=table-driven
```

```c
crc_t crc_update(crc_t crc, const void *data, size_t data_len)
{
    const unsigned char *d = (const unsigned char *)data;
    unsigned int tbl_idx;

    while (data_len--) {
        tbl_idx = crc ^ *d;
        crc = (crc_table[tbl_idx] ^ (crc >> 8)) & 0x1f;
        d++;
    }
    return crc & 0x1f;
}
```

I think the `& @crc_mask` should remain.

**Review by postmodern — CHANGES_REQUESTED** [review]
I compared your changes against [pycrc-0.10.0](https://pycrc.org/) which I'm using as a reference implementation, and found a few differences.

**Comment by postmodern** [comment]
> As a reminder, here is a brief summary of what is planned after this PR is merged:
> 
> * All `#updates` will be merged into `Digest::CRC#update`, so the `Digest::CRC5#update` that this PR modifies will also be removed.
> * The `Digest::CRC#pack` will also be merged, so the separate `#pack` method will be removed.
> * The C implementation also provides a general-purpose `Digest::CRC#update` method. Thus, specialized so files will only have initialization functions and table data.

Sounds good. Just as long as the classes that need to have custom/non-standard `update` or `pack` methods can still define there's, and we don't accidentally change the implementations of some of the more niche CRC algorithms. I assume `pack` will be replaced with some kind of `WIDTH` and a table of `Array#pack` codes? (ex: `{32 => 'L<', ...}`)

File: lib/digest/crc5.rb (hunk 1 above)
Comment (dearblue) [reply lib/digest/crc5.rb:15]: Hmmm, I understand. I will try to keep it in this PR.
However, it will eventually become unnecessary in later PRs.

**Pull request merged**